packages/provider/src/transports/base-wallet-transport.ts

```diff
@@ -375,6 +375,11 @@ export abstract class BaseWalletTransport implements WalletTransport {
       // Set connect options on the walletRequestHandler as our primary
       // wallet controller, and fall back to networkId if necessary
       this.walletRequestHandler.setConnectOptions(connectOptions)
+      if (connectOptions.networkId) {
+        networkId = connectOptions.networkId
+      } else if (networkId) {
+        connectOptions.networkId = networkId
+      }
```

Comment:
TBH I don't understand how setting `connectOptions.networkId` after `setConnectOptions` works, but that's how it was done before. Maybe we should consider moving this before `setConnectOptions`.